### PR TITLE
Add stft op

### PR DESCRIPTION
**Pull request by Lewuathe**
Add tf.signal.stft op. 

One TODO is passing fft length parameter because rfft does not support fft length parameter. We can pass fft length parameter after rfft supports it.

See: https://github.com/tensorflow/tfjs/issues/1362

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-core/1746)

### CODE REVIEW

**Comment by Lewuathe** [comment]
Okay, I will try to extend `rfft` ops.

**Comment by Lewuathe** [comment]
Updated rfft to get the fftLength parameter as TensorFlow core does.
See: https://github.com/tensorflow/tfjs-core/pull/1762

**Comment by Lewuathe** [comment]
@nsthorat I updated to support the given `fftLength`. Could you take a look when you get a chance?

**Comment by nsthorat** [comment]
Check out the details for failing tests:

image

**Comment by Lewuathe** [comment]
Oh, sorry. I'll take a look.

**Comment by Lewuathe** [comment]
Hmm, it keeps showing a weird error. That is exactly the same code used in the unit test. But it failed due to `tf.signal.stft is not a function.`

```
Step #6 - "test-snippets": ```js
Step #6 - "test-snippets": const input = tf.tensor1d([1, 1, 1, 1, 1])
Step #6 - "test-snippets": tf.signal.stft(input, 3, 1);
Step #6 - "test-snippets": ```
Step #6 - "test-snippets": 
Step #6 - "test-snippets": TypeError: tf.signal.stft is not a function
```

File: src/ops/signal_ops.ts [hunk 1]
```diff
@@ -107,3 +142,4 @@ function cosineWindow(windowLength: number, a: number, b: number): Tensor1D {
 export const hannWindow = op({hannWindow_});
 export const hammingWindow = op({hammingWindow_});
 export const frame = op({frame_});
+export const sftf = op({stft_});
```
Comment (nsthorat) [inline src/ops/signal_ops.ts]: the spelling is off here :)

**Comment by nsthorat** [comment]
Responded inline where the problem is.

File: src/ops/signal_ops.ts (hunk 1 above)
Comment (Lewuathe) [reply src/ops/signal_ops.ts]: Oops. Thanks for catching.

**Comment by nsthorat** [comment]
Stamped the complex64 change. Let's get this in ASAP if possible since we're going to move all the code in this repository to tensorflow/tfjs.

**Comment by manrajgrover** [comment]
@nsthorat But why? :open_mouth:

**Comment by nsthorat** [comment]
@manrajgrover having a monorepo will simplify a lot of the tooling we have, allowing us to share compiler settings, tslint settings, vscode settings, etc.

**Comment by nsthorat** [comment]
Also the test failure is a known flakiness issue that I fixed in https://github.com/tensorflow/tfjs-core/pull/1877 so just update to master.

**Comment by nsthorat** [comment]
Actually, for expediency, I'll merge this now and can you send a follow up to the monorepo once tfjs-core has moved?

**Comment by Lewuathe** [comment]
@nsthorat Sure, I'll make a follow-up PRs. Thanks!

**Pull request merged**